21.2.13 release notes (#14401)
cockroachdb/docs: _includes/releases/v21.2/v21.2.13.md

## v21.2.13

Release Date: July 5, 2022

{% include releases/release-downloads-docker-image.md release=include.release %}

<h3 id="v21-2-13-sql-language-changes">SQL language changes</h3>

- Fixed a small typo when using [DateStyle and IntervalStyle](../v21.2/set-vars.html#supported-variables). [#81547][#81547]
- Implemented the [`COPY FROM ... ESCAPE ...`](../v21.2/copy-from.html) syntax. [#82636][#82636]

> - Fixed an issue where [`SHOW BACKUP with privileges`](../v21.2/show-backup.html#show-a-backup-with-privileges) displayed grant statements with incorrect syntax (specifically, without the object type). As an example, previously displayed: `GRANT ALL ON status TO j4;`  Now displayed: `GRANT ALL ON TYPE status TO j4;`. [#82831][#82831]

Reviewer: ```suggestion
- Fixed an issue where [`SHOW BACKUP with privileges`](../v21.2/show-backup.html#show-a-backup-with-privileges) output grant statements with incorrect syntax (specifically, without the object type). For example, previously `SHOW BACKUP with privileges` output: `GRANT ALL ON status TO j4;` Now it correctly outputs: `GRANT ALL ON TYPE status TO j4;`. [#82831][#82831]
```

- In v21.1, a bug was introduced whereby default values were recomputed when populating data in new secondary indexes for columns which were added in the same transaction as the index. This would arise, for example, in cases like `ALTER TABLE t ADD COLUMN f FLOAT8 UNIQUE DEFAULT (random())`. If the default expression was not volatile, then the recomputation was harmless. If, however, the default expression was volatile, the data in the secondary index would not match the data in the primary index: a corrupt index would have been created. This bug has now been fixed. [#83223][#83223]
- Previously, a user could be connected to a database but unable to see the metadata for that database in `pg_catalog` if the user did not have privileges for the database. Now, users can always see the `pg_catalog` metadata for a database they are connected to. [#83359][#83359]
- Fixed a bug where it was possible to have a [virtual computed column](../v21.2/computed-columns.html) with an active `NOT NULL` constraint despite having rows in the table for which the column was `NULL`. [#83355][#83355]

> - Fix an issue with the [`soundex` function](../v21.2/functions-and-operators.html#built-in-functions) where certain unicode inputs could result in crashes, errors, or incorrect outputs. [#83434][#83434]

Reviewer: ```suggestion
- Fixed an issue with the [`soundex` function](../v21.2/functions-and-operators.html#built-in-functions) where certain unicode inputs could result in crashes, errors, or incorrect outputs. [#83434][#83434]
```
